pkg/token/tokendata.go

```diff
@@ -0,0 +1,64 @@
+// SPDX-FileCopyrightText: 2025 k0s authors
+// SPDX-License-Identifier: Apache-2.0
+
+package token
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/k0sproject/k0s/pkg/constant"
+)
+
+// GetTokenData resolves the join token from multiple possible sources:
+// CLI argument, token file, or K0S_TOKEN environment variable.
+// Returns the token data or an error if multiple sources are provided.
+// Returns empty string if no token source is available.
+func GetTokenData(tokenArg, tokenFile string) (string, error) {
+	tokenEnvValue := os.Getenv(constant.EnvVarToken)
+
+	tokenSources := 0
+	if tokenArg != "" {
+		tokenSources++
+	}
+	if tokenFile != "" {
+		tokenSources++
+	}
+	if tokenEnvValue != "" {
+		tokenSources++
+	}
+
+	if tokenSources > 1 {
+		return "", fmt.Errorf("you can only pass one token source: either as a CLI argument, via '--token-file [path]', or via the %s environment variable", constant.EnvVarToken)
```

Comment:
I wouldn't make the presence of the environment variable a hard error. It should rather be a fallback if the token hasn't been passed as an arg or via the file flag.

Reply:
It would be unclear in that case what token was used, do you want to add ambiguous behavior and solve that by extending documentation and telling people why passing token via env won't take precedence over a flag?